src/module.ts

```diff
@@ -114,6 +118,22 @@ export default defineNuxtModule<ModuleOptions>({
     nuxt.options.build.transpile.push(runtimeDir)
     addPlugin(resolve(runtimeDir, 'plugin'))
 
+    if (options.docSearch) {
+      const docSearchConfig = nuxt.options.publicRuntimeConfig.algolia.docSearch as Partial<DocSearchOptions>
```

Comment:
here there's going to be an error if a user sets the option to be `true` as it's not an object. I suggest moving this part before setting publicRuntimeConfig, making sure that `docSearch` is an object if it was set to true and then using `defu` for default values :)

Reply:
Hey @yassilah 🙂

I moved the check on top of runtimeConfig init, but kept the merging done by hand for two reasons:
- `defu` seemed overkill for such task 
- We only need to merge two keys that has inheritance from parent object to children
	- You can see previous discussions about this [here](https://github.com/nuxt-community/algolia-module/pull/42#discussion_r848177895)
	
@Baroshem; I also moved `runtimeDir` on top of the module declaration (has no side-effect) as I also moved the component injection to stay in the same `if` block for whole DocSearch feature.